source/tryGoal/component/tryGoalController.js

```diff
@@ -112,6 +113,31 @@ function($scope, agentMemory, $compile, $stateParams, config, GoalAPI, ConfigAPI
         $scope.$parent.splitScreen = !$scope.$parent.splitScreen;
     };
 
+    $scope.toggleQueryContext = function () {
+        $scope.$parent.displayQueryContext = !$scope.$parent.displayQueryContext;
+    }
+
+    function formatConcept(concept) {
+        if (concept === '(null)') {
+            return '(*)';
+        } else {
+            return concept;
+        }
+    };
+
+    $scope.formatRule = function (rule) {
+        var concepts = rule.match(/\((.*?)\)/gm);
+        var relationship = rule.match(/\[(.*?)\]/);
+
+        if (concepts && relationship) {
+            concepts[0] = formatConcept(concepts[0]);
+            concepts[1] = formatConcept(concepts[1]);
```

Comment:
Sorry, another question 😬 Will concepts and relationships always be of length 2 and will both [0] and [1] always have a value? It looks like you're handling '(null)' - is this the only case to worry about?

Reply:
The rule that's getting parsed is always in this format:
`"Inferring (Ellie)--[should eat at]->(null) (Rule 11/13)"`

There's always the subject and object ((Ellie) and (null)), the relationship ([should eat at]) and the rule number. The ticket describes that the only time we amend the output of one of the concepts is in the case of `null` where we make it a `*` to make it consistent with the training material. Can't think, off the top of my head, of any other times!